guillotina/asgi.py

```diff
@@ -0,0 +1,208 @@
+from aiohttp.streams import EmptyStreamReader
+from guillotina import glogging
+from guillotina import task_vars
+from guillotina.exceptions import ConflictError
+from guillotina.exceptions import TIDConflictError
+from guillotina.request import Request
+from typing import Optional
+
+import asyncio
+
+
+logger = glogging.getLogger('guillotina')
+
+
+def headers_to_list(headers):
+    return [[k.encode(), v.encode()] for k, v in headers.items()]
+
+
+class AsgiStreamReader(EmptyStreamReader):
```

Comment:
@dmanchon what do you think moving the reader to `request.py`? To avoid circular imports

Reply:
same, it makes sense to have all that logic together